core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -181,6 +185,34 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
       else consumer.pause(java.util.Collections.singleton(tp))
     }
 
+    def tryPoll(timeout: Long, onSuccess: ConsumerRecords[K, V] => Unit): Unit = {
+      Try(consumer.poll(timeout))
```

Comment:
This creates some allocations for each `tryPoll` and we should avoid that in tight loops like this. Can we write this with ordinary `try-catch` even if it might not look as elegant?